packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol

```diff
@@ -6,96 +6,104 @@ import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
- * The payable keyword is used on appendL2Output to save gas on the msg.value check.
- * This contract should be deployed behind an upgradable proxy
+ *         The payable keyword is used on appendL2Output to save gas on the msg.value check.
+ *         This contract should be deployed behind an upgradable proxy
  */
 // slither-disable-next-line locked-ether
 contract L2OutputOracle is Ownable {
+    /*********
+     * Types *
+     *********/
+
+    /**
+     * @notice OutputProposal represents a commitment to the L2 state.
+     *         The timestamp is the L1 timestamp that the output root is posted.
+     *         This timestamp is used to verify that the finalization period
+     *         has passed since the output root was submitted.
+     */
+    struct OutputProposal {
+        bytes32 outputRoot;
+        uint256 timestamp;
+    }
+
     /**********
      * Events *
      **********/
 
-    /// @notice Emitted when an output is appended.
+    /**
+     * @notice Emitted when an output is appended.
+     */
     event l2OutputAppended(
         bytes32 indexed _l2Output,
         uint256 indexed _l1Timestamp,
-        uint256 indexed _l2timestamp
+        uint256 indexed _l2BlockNumber
     );
 
-    /// @notice Emitted when an output is deleted.
+    /**
+     * @notice Emitted when an output is deleted.
+     */
     event l2OutputDeleted(
         bytes32 indexed _l2Output,
         uint256 indexed _l1Timestamp,
-        uint256 indexed _l2timestamp
+        uint256 indexed _l2BlockNumber
     );
 
     /**********************
      * Contract Variables *
      **********************/
 
-    /// @notice The interval in seconds at which checkpoints must be submitted.
+    /**
+     * @notice The interval in L2 blocks at which checkpoints must be submitted.
+     */
     uint256 public immutable SUBMISSION_INTERVAL;
 
-    /// @notice The time between blocks on L2.
-    uint256 public immutable L2_BLOCK_TIME;
-
-    /// @notice The number of blocks in the chain before the first block in this contract.
+    /**
+     * @notice The number of blocks in the chain before the first block in this contract.
+     */
     uint256 public immutable HISTORICAL_TOTAL_BLOCKS;
 
-    /// @notice The timestamp of the first L2 block recorded in this contract.
-    uint256 public immutable STARTING_BLOCK_TIMESTAMP;
+    /**
+     * @notice The number of the first L2 block recorded in this contract.
+     */
+    uint256 public immutable STARTING_BLOCK_NUMBER;
 
-    /// @notice The timestamp of the most recent L2 block recorded in this contract.
-    uint256 public latestBlockTimestamp;
+    /**
+     * @notice The number of the most recent L2 block recorded in this contract.
+     */
+    uint256 public latestBlockNumber;
 
-    /// @notice A mapping from L2 timestamps to the output root for the block with that timestamp.
+    /**
+     * @notice A mapping from L2 block numbers to the respective output root.
+     */
```

Comment:
Worth to add a warning that these proposals cannot be trusted until they are finalized (i.e. l1 timestamp of proposal must be old enough)